src/clp_ffi_js/ir/StreamReader.hpp

```diff
@@ -100,6 +115,108 @@ class StreamReader {
     [[nodiscard]] virtual auto decode_range(size_t begin_idx, size_t end_idx, bool use_filter) const
             -> DecodedResultsTsType = 0;
 
+    /**
```

Comment:
* `generic_filter_log_events` and `generic_decode_range` should be protected, right?
* Can we move the implementations after the class declaration?